### ISSUE TITLE
Custom formatter outside of fmt namespace with fmtlib>=9

### ISSUE THREAD

**Issue by eugene536** [fmtlib/fmt#3075]
Before `fmtlib9` we had a way to define `ostream& operator<<` to print custom types. Since version `9` such behavior is deprecated.

What extension points do we have now outside of `fmt` namespace?

The issue arise when we want to generate classes using macros in any namespace and don't want to wrap them with `fmt::streamable` all the time.

```
#define CUSTOM_STRUCT(name) \
struct name { friend std::ostream& operator<<(std::ostream& out, const name& v) { return out << 1; } };

CUSTOM_STRUCT(my_struct);

namespace foo {
  CUSTOM_STRUCT(my_struct);
}


int main() {
  fmt::print("{}", my_struct{});
  fmt::print("{}", foo::my_struct{});
}
```

example [on godbolt](https://godbolt.org/z/K4Y6W6TWM) that works fine with `FMT_DEPRECATED_OSTREAM` but I want a working solution without it.

**Comment by vitaut** [comment]
All extension points are documented in https://fmt.dev/latest/api.html#formatting-user-defined-types.

**Comment by eugene536** [comment]
@vitaut thanks for the reply, yeah I did this way, just added a tag to all generated classes and added formatter in `fmt` namespace for all such classes that contain that tag.

But the question was about extension points from another namespace and the answer as I see is there are no extension points in new versions of fmtlib outside of fmt :)

**Comment by vitaut** [comment]
Yeah, there are no such extension points other than `format_as` which is specific to enums.
